posthog/hogql/property.py

```diff
@@ -279,7 +279,7 @@ def property_to_expr(
             )[0:1].values_list("property_type", flat=True)
             property_type = property_types[0] if property_types else None
 
-            if not property_type or property_type == PropertyType.Boolean:
```

Comment:
Had to change this assumption for `retention.cy.ts` to pass, as the legacy demo data generator used by E2E tests doesn't add a property definition for this, hence the type is null – and we don't make that assumption in the spot where we wrap the `JSONExtractRaw` output in bool parsing, so a type mismatch was guaranteed @mariusandra